module/proto/gravitynft/v1/types.proto

```diff
@@ -13,9 +13,28 @@ message LastObservedNFTEthereumBlockHeight {
     uint64 ethereum_block_height = 2;
 }
 
-// TODO: Do we need UnhaltBridgeProposal?
+// This records the relationship between an ERC721 token and the class id
+// of the corresponding Cosmos originated asset
+message ERC721ToClassId {
+    string erc721 = 1;
+    string class_id = 2;
+  }
 
-// TODO: Do we need IBCMetadataProposal?
```

Comment:
IBC denoms for tokens are constructed automatically, but for creating NFT's class ids it's on us right? So aren't we going to set exactly what we need on ibc receive?

Reply:
So, the description from the erc20 part is:
```
// IBCMetadataProposal defines a custom governance proposal type that allows governance to set the
// metadata for an IBC token, this will allow Gravity to deploy an ERC20 representing this token on
// Ethereum
```

I wasn't aware that this was even needed, but I see they are used for any cosmos-originated tokens like this: https://www.mintscan.io/gravity-bridge/proposals/179

So you are saying you think we won't need this?